autometa/config/databases.py

```diff
@@ -111,185 +232,388 @@ def format_nr(self):
 
-                all available evaluates to True, otherwise False
+        Parameters
+        ----------
+        section : str, optional
+            Section to check for missing database files (the default is None).
+            Choices include 'ncbi' and 'markers'.
 
-            - if `validate` is False : dict
+        Returns
+        -------
+        NoneType
+            Will update provided `section` in `self.config`.
 
-                {section:{option, option,...}, section:{...}, ...}
+        Raises
+        -------
+        ValueError
+            Provided `section` does not match 'ncbi' and 'markers'.
 
         """
-        missing = {}
-        for section in Databases.SECTIONS:
+        dispatcher = {'ncbi': self.download_ncbi_files,
+                      'markers': self.download_markers}
+        if section and section not in dispatcher:
+            raise ValueError(f'{section} does not match "ncbi" or "markers"')
+        if section:
+            missing = self.get_missing(section=section)
+            options = missing.get(section, [])
+            dispatcher[section](options)
+        else:
+            missing = self.get_missing()
+            for section, options in missing.items():
+                dispatcher[section](options)
+
+    def compare_checksums(self, section=None):
+        """Get all invalid database files in `options` from `section`
+        in config. An md5 checksum comparison will be performed between the
+        current and file's remote md5 to ensure file integrity prior to checking
+        the respective file as valid.
+
+        Parameters
+        ----------
+        section : str, optional
+            Configure provided `section`. Choices include 'markers' and 'ncbi'.
+            (default will download/format all database directories)
+
+        Returns
+        -------
+        dict
+            {section:{option, option,...}, section:{...}, ...}
+
+        """
+        sections = [section] if section else Databases.SECTIONS.keys()
+        invalid = {}
+        taxdump_checked = False
+        for section in sections:
             for option in self.config.options(section):
                 if option not in Databases.SECTIONS.get(section):
                     # Skip user added options not required by Autometa
                     continue
+                # nodes.dmp, names.dmp and merged.dmp are all in taxdump.tar.gz
+                option = 'taxdump' if option in {
+                    'nodes', 'names', 'merged'} else option
                 fpath = self.config.get(section, option)
-                if os.path.exists(fpath) and os.stat(fpath).st_size >= 0:
-                    # TODO: [Checkpoint validation]
-                    logger.debug(f'({section},{option}): {fpath}')
+                fpath_md5 = f'{fpath}.md5'
+                # We can not checksum a file that does not exist.
+                if not os.path.exists(fpath) and not os.path.exists(fpath_md5):
```

Comment:
If `fpath` exists but `fpath_md5` doesn't exist, doesn't that suggest some problem occurred during download? More to the point, should we do something in this case rather than skipping with `continue`?

Reply:
This only skips if _both_ the `fpath` and the `fpath_md5` do not exist